contracts/tips/src/contract.rs

```diff
@@ -0,0 +1,2202 @@
+        tip.ref_counter -= 1;
+        if tip.ref_counter == 0 {
+            // Ref counter is zero, we can safely remove the tip from the storage
+            TIPS.remove(storage, removed_tip_key);
+        } else {
+            // Not zero, update the tip with the new ref counter
+            TIPS.save(storage, removed_tip_key, &tip)?;
+        }
+    }
+
+    history_map.save(storage, subject, &subject_history)
+}
+
+fn execute_update_service_fee(
+    deps: DepsMut<DesmosQuery>,
+    info: MessageInfo,
+    service_fee: Option<ServiceFee>,
+) -> Result<Response<DesmosMsg>, ContractError> {
+    let mut config = CONFIG.load(deps.storage)?;
+
+    if config.admin != info.sender {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    let new_service_fee = if let Some(service_fee) = service_fee {
+        Some(StateServiceFee::try_from(service_fee)?)
+    } else {
+        None
+    };
+    config.service_fee = new_service_fee;
+    CONFIG.save(deps.storage, &config)?;
+
+    Ok(Response::new()
+        .add_attribute(ATTRIBUTE_ACTION, ACTION_UPDATE_SERVICE_FEE)
+        .add_attribute(ATTRIBUTE_SENDER, info.sender))
+}
+
+fn execute_update_admin(
+    deps: DepsMut<DesmosQuery>,
+    info: MessageInfo,
+    new_admin: String,
+) -> Result<Response<DesmosMsg>, ContractError> {
+    let new_admin = deps.api.addr_validate(&new_admin)?;
+    CONFIG.update(deps.storage, |mut config| {
+        if config.admin != info.sender {
+            return Err(ContractError::Unauthorized {});
+        }
+        config.admin = new_admin.clone();
+        Ok(config)
+    })?;
+
+    Ok(Response::new()
+        .add_attribute(ATTRIBUTE_ACTION, ACTION_UPDATE_ADMIN)
+        .add_attribute(ATTRIBUTE_SENDER, info.sender)
+        .add_attribute(ATTRIBUTE_NEW_ADMIN, new_admin))
+}
+
+fn execute_update_saved_tips_history_size(
+    deps: DepsMut<DesmosQuery>,
+    info: MessageInfo,
+    new_size: u32,
+) -> Result<Response<DesmosMsg>, ContractError> {
+    let mut config = CONFIG.load(deps.storage)?;
+
+    if config.admin != info.sender {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    // Wipe the tips history otherwise leave to SendTip to shrink the tips record
+    if new_size == 0 {
+        wipe_map(&TIPS, deps.storage)?;
+        wipe_map(&SENT_TIPS_HISTORY, deps.storage)?;
+        wipe_map(&RECEIVED_TIPS_HISTORY, deps.storage)?;
+        wipe_map(&POST_TIPS_HISTORY, deps.storage)?;
+    }
+
+    config.tips_history_size = new_size;
+    CONFIG.save(deps.storage, &config)?;
+
+    Ok(Response::new()
+        .add_attribute(ATTRIBUTE_ACTION, ACTION_UPDATE_SAVED_TIPS_HISTORY_SIZE)
+        .add_attribute(ATTRIBUTE_SENDER, info.sender)
+        .add_attribute(ATTRIBUTE_NEW_SIZE, new_size.to_string()))
+}
+
+/// Removes all the items inside a [`cw_storage_plus::Map`].
+/// * `map` - Map from where will be removed all the items.
+fn wipe_map<'a, K, V>(map: &Map<'a, K, V>, storage: &mut dyn Storage) -> StdResult<()>
```

Comment:
I was thinking...could this be a good thing to add to the `cw_storage_plus::Map` through a PR contribution? It looks like something that would be useful for any dev working with CW Maps

Reply:
Plus, I don't see any test for this function. Can we add some?

Reply:
Yes i can open an issue to start a discussion with the CosmWASM team about that.
About the tests this funtion don't have a dedicated test but it's correctnes is covered inside [this test](https://github.com/desmos-labs/desmos-contracts/blob/b78f182de299cf0a0b3fde5e8d1d4fd0e421b918/contracts/tips/src/contract.rs#L1699)

Reply:
Awesome for both!